diff --git a/src/vulkan/meson.build b/src/vulkan/meson.build
--- a/src/vulkan/meson.build
+++ b/src/vulkan/meson.build
@@ -68,11 +68,13 @@ if with_platform_windows
   vulkan_wsi_list += '-DVK_USE_PLATFORM_WIN32_KHR'
 endif
 if with_platform_macos
   vulkan_wsi_list += '-DVK_USE_PLATFORM_MACOS_MVK'
+endif
+if with_platform_macos or with_platform_ios
   vulkan_wsi_list += '-DVK_USE_PLATFORM_METAL_EXT'
   vulkan_wsi_deps += dependency(
     'appleframeworks',
     modules : ['QuartzCore', 'Metal']
   )
 endif
 
diff --git a/src/vulkan/wsi/meson.build b/src/vulkan/wsi/meson.build
--- a/src/vulkan/wsi/meson.build
+++ b/src/vulkan/wsi/meson.build
@@ -34,7 +34,7 @@ else
   files_vulkan_wsi += files('wsi_common_headless.c')
 endif
 
-if with_platform_macos
+if with_platform_macos or with_platform_ios
   files_vulkan_wsi += files('wsi_common_metal.c', 'wsi_common_metal_layer.m')
 endif
 

diff --git a/src/vulkan/wsi/wsi_common_metal_layer.m b/src/vulkan/wsi/wsi_common_metal_layer.m
--- a/src/vulkan/wsi/wsi_common_metal_layer.m
+++ b/src/vulkan/wsi/wsi_common_metal_layer.m
@@ -9,6 +9,7 @@
 
 #import <QuartzCore/CAMetalLayer.h>
 #import <Metal/Metal.h>
+#import <TargetConditionals.h>
 
 void
 wsi_metal_layer_size(const CAMetalLayer *metal_layer,
@@ -139,7 +140,12 @@ void
 wsi_metal_layer_set_immediate(const CAMetalLayer *metal_layer,
                               bool enable_immediate)
 {
+#if TARGET_OS_OSX
    metal_layer.displaySyncEnabled = !enable_immediate;
+#else
+   (void)metal_layer;
+   (void)enable_immediate;
+#endif
 }
 
 VkResult
@@ -170,7 +176,9 @@ wsi_metal_layer_configure(const CAMetalLayer *metal_layer,
       metal_layer.maximumDrawableCount = image_count;
       metal_layer.drawableSize = (CGSize){.width = width, .height = height};
       metal_layer.opaque = enable_opaque;
+#if TARGET_OS_OSX
       metal_layer.displaySyncEnabled = !enable_immediate;
+#endif
       metal_layer.pixelFormat = metal_format;
 
       metal_layer.colorspace = cg_color_space;
